physionet-django/project/management/commands/get_metrics_data.py

```diff
@@ -0,0 +1,267 @@
+                if slug not in data:
+                    data[slug] = [date, 0, set()]
+                if ip not in data[slug][2]:
+                    data[slug][1] += 1
+                    data[slug][2].add(ip)
+        except IndexError:
+            if line:
+                print("Invalid line in log:", line)
+    return data
+
+
+def update_metrics(filename):
+    """Updates project metrics in the database.
+
+    Obtains metrics data from log_parser and then updates each core project
+    with new data.
+
+    Args:
+        filename: An NGINX log file to parse.
+
+    Returns:
+        None
+    """
+    log_data = log_parser(filename)
+
+    for p in PublishedProject.objects.filter(is_latest_version=True):
+        if p.slug in log_data:
+            try:
+                last_entry = Metrics.objects.filter(
+                    core_project=p.core_project).latest('date')
+                # Do not update if log file is older than latest entry
+                if last_entry.date >= log_data[p.slug][0].date():
+                    continue
+            except ObjectDoesNotExist:
+                last_entry = None
+            project = Metrics.objects.create(
+                core_project=p.core_project,
+                slug=p.slug,
+                date=log_data[p.slug][0])
+            if last_entry:
+                project.running_viewcount = last_entry.running_viewcount
+            project.viewcount = log_data[p.slug][1]
+            project.running_viewcount += log_data[p.slug][1]
+            project.save()
+
+
+def validate_log_file(filename, is_cron=False, now=None):
+    """Checks if a log file is a valid NGINX log and has correct date
+
+    Checks if the given file has a date in the right place and format. If
+        called during a cron job, checks if the log file's date is the day
+        before the current date.
+
+    Args:
+        filename: An NGINX log file to parse.
+        is_cron: Optional; If True, checks if date is correct for cron job.
+        now: Optional; If running cron job, treats this as the currrent date
+            instead of timezone.now().
+
+    Returns:
+        None
+
+    Raises:
+        InvalidLogError: An error occurred obtaining date data from the log
+            file.
+        DateError: The date in the log file does not match the expected date
+            (if run as a cron job, the log file date should be the day before).
+    """
+    if now is None:
+        now = timezone.now()
+
+    with open(filename) as f:
+        first_line = f.readline()
+        split_line = first_line.split()
+        try:
+            str_date = split_line[3][1:12]
+            log_date = str_to_datetime(str_date)
+        except LookupError:
+            raise InvalidLogError
+
+    if is_cron:
+        if not (now.day - 1 == log_date.day and now.month ==
+                log_date.month and now.year == log_date.year):
+            raise DateError
+    return
+
+
+def str_to_datetime(str_date):
```

Comment:
Could this function be replaced with `datetime.strptime`?

e.g. 

```
from datetime import datetime
date = '20/10/20'
date_object = datetime.strptime(date, "%d/%m/%y")
```

Reply:
Yes thank you, that works great!